docs/content/development/extensions-core/druid-kerberos.md

```diff
@@ -0,0 +1,70 @@
+---
+layout: doc_page
+---
+
+# Druid-Kerberos
+
+Druid Extension to enable Authentication for Druid Nodes using Kerberos.
+This extension adds AuthenticationFilter which is used to protect HTTP Endpoints using the simple and protected GSSAPI negotiation mechanism [SPNEGO](https://en.wikipedia.org/wiki/SPNEGO). 
+Make sure to [include](../../operations/including-extensions.html) `druid-kerberos` as an extension.
+
+
+## Configuration
+
+|Property|Possible Values|Description|Default|required|
+|--------|---------------|-----------|-------|--------|
+|`druid.hadoop.security.kerberos.principal`|`druid@EXAMPLE.COM`| Principal user name, used for internal node communication|empty|Yes|
+|`druid.hadoop.security.kerberos.keytab`|`/etc/security/keytabs/druid.headlessUser.keytab`|Path to keytab file used for internal node communication|empty|Yes|
+|`druid.hadoop.security.spnego.principal`|`HTTP/_HOST@EXAMPLE.COM`| SPNego service principal used by druid nodes|empty|Yes|
+|`druid.hadoop.security.spnego.keytab`|`/etc/security/keytabs/spnego.service.keytab`|SPNego service keytab used by druid nodes|empty|Yes|
+|`druid.hadoop.security.spnego.authToLocal`|`RULE:[1:$1@$0](druid@EXAMPLE.COM)s/.*/druid DEFAULT`|It allows you to set a general rule for mapping principal names to local user names. It will be used if there is not an explicit mapping for the principal name that is being translated.|DEFAULT|No|
+|`druid.hadoop.security.spnego.excludedPaths`|`['/status','/health']`| Array of HTTP paths which which does NOT need to be authenticated.|None|No|
+|`druid.hadoop.security.spnego.cookieSignatureSecret`|`secretString`| Secret used to sign authentication cookies|<Random value>|No|
```

Comment:
main reason is that the cookie specs does not guarantee isolation against port numbers. 
e.g This is a sample cookie content from my host which does not have any port info- 
```
druid-spnego-1.openstacklocal	FALSE	/druid/v2/	FALSE	0	hadoop.auth	u=druid&p=druid@EXAMPLE.COM&t=kerberos&e=1486006717274&s=/pK2gDY6htUnS73Bo2DrwDSTh2Y=
```
 so if you have multiple druid nodes running on same host using same paths and different signature (generated randomly instead of being explictly set ), user will keep doing token negotiation as cookie generated by one node will not be accepted by other node. setting the cookie signature ensures that this does not occur. 
Have added more clarification in docs.